### PR TITLE
Break Play WS into modules

### PR DESCRIPTION
**Pull request by wsargent**
This PR breaks apart WS functionality in the following ways:

* Moves the AHC implementation from play-ws to play-ahc-ws.
* Remove play-java-ws and moves the Java classes to play-ws and play-ahc-ws.
* Moves the OpenID classes in the Play-WS package out to a new module play-openid.

There is intentionally no code changes (including the ugly Netty dependency in playOpenIdDeps) so they can be tackled in subsequent PRs and the git log will be easier to follow i.e. rename and repackage first, then start tweaking.

### CODE REVIEW

File: framework/project/Dependencies.scala [hunk 1]
```diff
@@ -256,6 +256,11 @@ object Dependencies {
     (specsBuild :+ specsMatcherExtra).map(_ % Test) :+
     mockitoAll % Test
 
+  val playOpenIdDeps = Seq(
+    // import org.jboss.netty.handler.codec.http.QueryStringDecoder in package.scala
+    "io.netty" % "netty" % "3.10.6.Final" % Test
```
Comment (schmitch) [inline framework/project/Dependencies.scala]: this is a nasty one, can't we just copy that class?

Comment (wsargent) [inline framework/project/Dependencies.scala]: we will.  I'm going to do that in another PR though so it flows better.

**Comment by ktoso** [comment]
yay!

**Comment by wsargent** [comment]
@ktoso if you liked this, you're going to love the next PR I've got lined up ;-)

**Comment by ktoso** [comment]
Oh my gawd, the suspense is killing me! :D
